src/lib.d

```diff
@@ -10,44 +10,47 @@ module ddmd.lib;
 
 import ddmd.globals;
 
-static if (TARGET_WINDOS)
+// Include all of these once these modules have been ported to all platforms.
+version (Windows)
 {
     import ddmd.libomf;
     import ddmd.libmscoff;
 }
-else static if (TARGET_LINUX || TARGET_FREEBSD || TARGET_OPENBSD || TARGET_SOLARIS)
+else version (OSX)
 {
-    import ddmd.libelf;
+    import ddmd.libmach;
 }
-else static if (TARGET_OSX)
+else version (Posix)
 {
-    import ddmd.libmach;
+    import ddmd.libelf;
 }
 else
 {
-    static assert(0, "unsupported system");
+    static assert(0, "OS not supported.");
 }
 
 extern (C++) class Library
 {
 public:
     static Library factory()
     {
-        static if (TARGET_WINDOS)
+        // This should be turned into runtime logic based on TARGET_* once
+        // these library generation modules are made cross-platform.
+        version (Windows)
         {
             return (global.params.mscoff || global.params.is64bit) ? LibMSCoff_factory() : LibOMF_factory();
         }
-        else static if (TARGET_LINUX || TARGET_FREEBSD || TARGET_OPENBSD || TARGET_SOLARIS)
```

Comment:
The same would be true for the rest of the uses of `TARGET_POSIX` in this PR, unless you're saying that Posix is more likely to use the .o and .a extensions set in `ddmd.globals` than use ELF?  I don't see how that's okay, but this isn't.

I'm well aware that `version (OSX)` is not the same as `TARGET_OSX`, which is why I mentioned that this should be changed to `TARGET_OSX` eventually in the comment I added above.  The issue is that these factories are written only to compile _on_ the target platform, because they include druntime headers that define constants only on an `OSX` or `Windows` host, an ironic consequence of going ddmd.

If somebody wants to make dmd into a cross-compiler, they'll need to make these modules cross-platform and use `TARGET_OSX` instead.  Until then, this is a hack to avoid that portability issue, hence the comment.